examples/dev-prod/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fmt::{self},
+    str::FromStr,
+};
+
+#[derive(PartialEq)]
+enum AppEnv {
+    Dev,
+    Prod,
+}
+
+/// A common setup that:
+///  - loads from a .env file in dev mode
+///  - loads from the envrironment in prod mode
+///
+/// A few commands to try:
+/// 1) `APP_ENV=prod HOST=prod.com cargo run`
+/// 2) `APP_ENV=dev HOST=prod.com cargo run`
+/// 3) `APP_ENV=prod cargo run`
+///
+/// To have the .env file take priority, use `dotenv_orderride()`.
+/// Try replacing `dotenv()` with `dotenv_override()` on line 2 and re-running command 2.
+fn main() {
+    let app_env = std::env::var("APP_ENV")
+        .unwrap_or("dev".to_owned())
+        .parse::<AppEnv>()
+        .unwrap();
```

Comment:
Could remove the `FromStr` implementation and change this to:

```suggestion
    let app_env = match env::var("APP_ENV") {
        Ok(val) if val == "prod" => AppEnv::Prod,
        _ => AppEnv::Dev,
    };
```

Would simplify the example, but doesn't account for the case that a weird APP_ENV is set. Might be out of the scope of this demonstration.

Reply:
Yes, yours reads better top-to-bottom (which is good for an example).

Incorporated